host/async_wf_test.go

```diff
@@ -0,0 +1,248 @@
+		name             string
+		wantStartFailure bool
+		asyncWFCfg       *types.AsyncWorkflowConfiguration
+	}{
+		{
+			name:             "start workflow execution async fails because domain missing async queue",
+			wantStartFailure: true,
+		},
+		{
+			name: "start workflow execution async fails because async queue is disabled",
+			asyncWFCfg: &types.AsyncWorkflowConfiguration{
+				Enabled: false,
+			},
+			wantStartFailure: true,
+		},
+		{
+			name: "start workflow execution async succeeds and workflow starts",
+			asyncWFCfg: &types.AsyncWorkflowConfiguration{
+				Enabled:             true,
+				PredefinedQueueName: "test-async-wf-queue",
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+		s.T().Run(tc.name, func(t *testing.T) {
+			// advance the time so each test has a unique start time
+			s.testClusterConfig.TimeSource.Advance(time.Second)
+
+			ctx, cancel := createContext()
+			defer cancel()
+
+			if tc.asyncWFCfg != nil {
+				_, err := s.adminClient.UpdateDomainAsyncWorkflowConfiguraton(ctx, &types.UpdateDomainAsyncWorkflowConfiguratonRequest{
+					Domain:        s.domainName,
+					Configuration: tc.asyncWFCfg,
+				})
+				if err != nil {
+					t.Fatalf("UpdateDomainAsyncWorkflowConfiguraton() failed: %v", err)
+				}
+
+				s.domainCacheRefresh()
+			}
+
+			startTime := s.testClusterConfig.TimeSource.Now().UnixNano()
+			wfID := fmt.Sprintf("async-wf-integration-start-workflow-test-%d", startTime)
+			wfType := "async-wf-integration-start-workflow-test-type"
+			taskList := "async-wf-integration-start-workflow-test-tasklist"
+			identity := "worker1"
+
+			asyncReq := &types.StartWorkflowExecutionAsyncRequest{
+				StartWorkflowExecutionRequest: &types.StartWorkflowExecutionRequest{
+					RequestID:  uuid.New(),
+					Domain:     s.domainName,
+					WorkflowID: wfID,
+					WorkflowType: &types.WorkflowType{
+						Name: wfType,
+					},
+					TaskList: &types.TaskList{
+						Name: taskList,
+					},
+					Input:                               nil,
+					ExecutionStartToCloseTimeoutSeconds: common.Int32Ptr(100),
+					TaskStartToCloseTimeoutSeconds:      common.Int32Ptr(1),
+					Identity:                            identity,
+				},
+			}
+
+			_, err := s.engine.StartWorkflowExecutionAsync(ctx, asyncReq)
+			if tc.wantStartFailure != (err != nil) {
+				t.Errorf("StartWorkflowExecutionAsync() failed: %v, wantStartFailure: %v", err, tc.wantStartFailure)
+			}
+
+			if err != nil || tc.wantStartFailure {
+				return
+			}
+
+			// there's no worker or poller for async workflow, so we just validate whether it started.
+			// this is sufficient to verify the async workflow start path.
+			for i := 0; i < 1000; i++ {
+				timeFilter := &types.StartTimeFilter{
+					EarliestTime: common.Int64Ptr(time.Unix(0, 0).UnixNano()),
+					LatestTime:   common.Int64Ptr(s.testClusterConfig.TimeSource.Now().UnixNano()),
+				}
+				resp, err := s.engine.ListOpenWorkflowExecutions(ctx, &types.ListOpenWorkflowExecutionsRequest{
```

Comment:
Better to use `DescribeWorkflowExecution` API to verify, because it doesn't depend on visibility index.